cpp/src/io/text/data_chunk_source_factories.cpp

```diff
@@ -87,8 +87,10 @@ class datasource_chunk_reader : public data_chunk_reader {
       _source->host_read(_offset, read_size, reinterpret_cast<uint8_t*>(h_ticket.buffer.data()));
 
       // copy the host-pinned data on to device
-      CUDF_CUDA_TRY(cudaMemcpyAsync(
-        chunk.data(), h_ticket.buffer.data(), read_size, cudaMemcpyDefault, stream.value()));
+      cudf::detail::cuda_memcpy_async<char>(
```

Comment:
There are a number of places where the template argument (char) is given explcitly ... is the compiler really not able to deduce it from the inputs?

Reply:
hm, maybe it wasn't required here
In general, compiler can't do template type deduction + implicit conversion. So passing a container that get implicitly converted to a span requires the template type for `cuda_memcpy_async`.